### PR TITLE
Modernize server.js

### PR DESCRIPTION
**Pull request by jimmywarting**
modernize server.js

Uses classes, const let etc
Can no longer create a server without `new` keyword

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -36,745 +31,720 @@ inherits(Server, EventEmitter)
...
-      self.http.on('request', function (req, res) {
-        if (res.headersSent) return
-        self.onHttpRequest(req, res)
-      })
-    })
-  }
+class Server extends EventEmitter {
+  constructor (opts = {}) {
+    super()
+    const self = this
```
Comment (DiegoRBaquero) [inline server.js]: We could get rid of self

File: server.js [hunk 2]
```diff
@@ -36,745 +31,720 @@ inherits(Server, EventEmitter)
...
-            '<h3>IPv4 Peers: ' + stats.peersIPv4 + '</h3>\n' +
-            '<h3>IPv6 Peers: ' + stats.peersIPv6 + '</h3>\n' +
-            '<h3>Clients:</h3>\n' +
-            printClients(stats.clients)
-          )
+          if (req.url === '/stats.json' || req.headers['accept'] === 'application/json') {
+            res.write(JSON.stringify(stats))
+            res.end()
+          } else if (req.url === '/stats') {
+            res.end(`<h1>${stats.torrents} torrents (${stats.activeTorrents} active)</h1>\n<h2>Connected Peers: ${stats.peersAll}</h2>\n<h3>Peers Seeding Only: ${stats.peersSeederOnly}</h3>\n<h3>Peers Leeching Only: ${stats.peersLeecherOnly}</h3>\n<h3>Peers Seeding & Leeching: ${stats.peersSeederAndLeecher}</h3>\n<h3>IPv4 Peers: ${stats.peersIPv4}</h3>\n<h3>IPv6 Peers: ${stats.peersIPv6}</h3>\n<h3>Clients:</h3>\n${printClients(stats.clients)}`
```
Comment (DiegoRBaquero) [inline server.js]: Much better!

Comment (DiegoRBaquero) [reply server.js]: I think we can remove the `\n`, as header tags do break lines.

File: server.js (hunk 1 above)
Comment (jimmywarting) [reply server.js]: done

File: server.js (hunk 2 above)
Comment (jimmywarting) [reply server.js]: solved

**Pull request merged**